arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceOptionsViewController.swift

```diff
@@ -0,0 +1,279 @@
+            object: nil,
+            queue: .main
+        ) { _ in
+            if let text = alertController.textFields!.first!.text {
+                // Enable the done button if the textfield is not empty and is a valid number.
+                doneAction.isEnabled = NumberFormatter().number(from: text) != nil
+            } else {
+                doneAction.isEnabled = false
+            }
+        }
+        // Add a cancel action to alert controller.
+        alertController.addAction(cancelAction)
+        alertController.preferredAction = doneAction
+        present(alertController, animated: true)
+    }
+    
+    // MARK: UITableViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        selectedAttribute = nil
+        selectedComparison = nil
+        selectedValue = nil
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        let cell = tableView.cellForRow(at: indexPath)
+        switch cell {
+        case attributesCell:
+            showAttributePicker()
+        case comparisonCell:
+            showComparisonPicker()
+        case valueCell:
+            if let domain = selectedAttribute?.domain as? AGSCodedValueDomain {
+                showValuePicker(values: domain.codedValues)
+            } else {
+                showValueInputField { [weak self] value in
+                    guard let self = self else { return }
+                    // Assign an `NSNumber?` to selected value so that it can cast to numbers.
+                    self.selectedValue = value
+                    self.valueCell.detailTextLabel?.text = value?.stringValue
+                    // Mitigate the Apple's UI bug in right detail cell.
+                    tableView.reloadRows(at: [indexPath], with: .none)
+                }
+            }
+        default:
+            fatalError("Unknown cell type")
+        }
+    }
+}
+
+/// An extension of `AGSUtilityAttributeComparisonOperator` that returns a human readable description.
+private extension AGSUtilityAttributeComparisonOperator {
+    static let allCases = (Self.equal.rawValue...Self.doesNotIncludeAny.rawValue).map { AGSUtilityAttributeComparisonOperator(rawValue: $0)! }
+    
+    var title: String {
+        switch self {
+        case .equal: return "Equal"
+        case .notEqual: return "NotEqual"
+        case .greaterThan: return "GreaterThan"
+        case .greaterThanEqual: return "GreaterThanEqual"
+        case .lessThan: return "LessThan"
+        case .lessThanEqual: return "LessThanEqual"
+        case .includesTheValues: return "IncludesTheValues"
+        case .doesNotIncludeTheValues: return "DoesNotIncludeTheValues"
+        case .includesAny: return "IncludesAny"
+        case .doesNotIncludeAny: return "DoesNotIncludeAny"
+        @unknown default: return "Unknown"
```

Comment:
Should these titles be more natural? For example, "Not Equal" instead of "NotEqual"?